src/util/git.js

```diff
@@ -33,6 +37,27 @@ const handleSpawnError = err => {
   }
 };
 
+const SHORTHAND_SERVICES: {[key: string]: url.parse} = map({
+  'github:': parsedUrl => ({
+    ...parsedUrl,
+    slashes: true,
+    auth: 'git',
+    protocol: SSH_PROTOCOL,
+    host: 'github.com',
+    hostname: 'github.com',
+    pathname: `/${parsedUrl.hostname}${parsedUrl.pathname}`,
+  }),
+  'bitbucket:': parsedUrl => ({
+    ...parsedUrl,
+    slashes: true,
+    auth: 'git',
+    protocol: SSH_PROTOCOL,
+    host: 'bitbucket.com',
+    hostname: 'bitbucket.com',
+    pathname: `/${parsedUrl.hostname}${parsedUrl.pathname}`,
+  }),
```

Comment:
There are 2 questions here so I'll answer them separately :)

1. Why is GitLab one of them?

Because it wasn't there already and this is just a refactor so we can add them easily in a follow-up feature patch :)

2. Why don't we generalize it more?

I thought about a similar pattern but it makes too many assumptions about the structure like what and how `auth` should be or how the pathname should be expanded and your example even abstracts out the hostname part :D We may go there in the future probably but I don't want to overly restrict this part.

As an addendum to this, I considered having custom expansions through `.yarnrc` file, something that would translate into
```
shorthandServices: {
    github: {
        auth: 'git',
        hostname: 'github.com',
    }
}
```

This way we can let people define these themselves externally (even with plugin packages may be?). If we do this in the yarnrc file, it will be more limited, if we allow plugins, it would be a lot more flexible but potentially complex, since plugins. I still think people may find this useful defining internal company shortcuts like SSH host aliases.

TL;DR
Let's keep it a bit more flexible as it is to keep us open to new opportunities?

Reply:
thanks for explaining :)

Reply:
Thanks a lot for asking since I was able to do a brain dump. Any thoughts on them?

Reply:
I like the idea of putting it in `.yarnrc` -- i'd imagine most people would just need to configure host or hostname (maybe protocol?) and don't need to deal with the overhead of plugins beyond that.